arkane/mainTest.py

```diff
@@ -0,0 +1,110 @@
+# RMG - Reaction Mechanism Generator                                          #
+#                                                                             #
+# Copyright (c) 2002-2018 Prof. William H. Green (whgreen@mit.edu),           #
+# Prof. Richard H. West (r.west@neu.edu) and the RMG Team (rmg_dev@mit.edu)   #
+#                                                                             #
+# Permission is hereby granted, free of charge, to any person obtaining a     #
+# copy of this software and associated documentation files (the 'Software'),  #
+# to deal in the Software without restriction, including without limitation   #
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,    #
+# and/or sell copies of the Software, and to permit persons to whom the       #
+# Software is furnished to do so, subject to the following conditions:        #
+#                                                                             #
+# The above copyright notice and this permission notice shall be included in  #
+# all copies or substantial portions of the Software.                         #
+#                                                                             #
+# THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR  #
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,    #
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE #
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER      #
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING     #
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER         #
+# DEALINGS IN THE SOFTWARE.                                                   #
+#                                                                             #
+###############################################################################
+
+"""
+This script contains unit tests of the :mod:`arkane.main` module.
+"""
+
+import unittest
+import os
+import shutil
+
+import rmgpy
+from arkane import Arkane
+
+################################################################################
+
+
+class TestArkaneExamples(unittest.TestCase):
+    """
+    Run all of Arkane's examples, and report which one failed
+    """
+    @classmethod
+    def setUp(self):
+        """A function that is run ONCE before all unit tests in this class."""
+        self.base_path = os.path.join(os.path.dirname(os.path.dirname(rmgpy.__file__)), 'examples', 'arkane')
+        self.failed = []
+        self.example_types = ['species', 'reactions', 'networks']
+
+    def test_arkane_examples(self):
+        for example_type in self.example_types:
+            example_type_path = os.path.join(self.base_path, example_type)
+            examples = []
+            for (dirpath, dirnames, filenames) in os.walk(example_type_path):
+                examples.extend(dirnames)
```

Comment:
I think you can use `os.listdir(example_type_path)` here since all the examples are in subdirectories.